internal/pkg/service/buffer/worker/service/cleanup.go

```diff
@@ -59,8 +56,11 @@ func (s *Service) cleanupTasks(ctx context.Context, wg *sync.WaitGroup, d depend
 			case <-ctx.Done():
 				return
 			case <-ticker.C:
-				if err := node.Check(ctx); err != nil && !errors.Is(err, context.Canceled) {
-					logger.Error(err)
+				// Only one worker should do cleanup
+				if s.dist.MustCheckIsOwner("task.cleanup") {
+					if err := s.tasks.Cleanup(); err != nil && !errors.Is(err, context.Canceled) {
+						logger.Error(err)
+					}
 				}
```

Comment:
Prvy commit, uprava services, co chceme dosiahnut?

`task.Node.Cleanup`, nepotrebujeme tu vytvarat nejaku "cleanup node", zavislosti su rovnake.

Mozeme pouzit `s.dist.MustCheckIsOwner("task.cleanup") `, aj ked je tam aj zamok, aby to bezalo iba raz.